File: framework/csharp/soda/unittest/WorkOutput.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;

namespace Soda.Unittest;

public class WorkOutput
{
    [JsonProperty("id")]
    public int Id { get; set; }

    [JsonProperty("success")]
    public bool Success { get; set; }

    [JsonProperty("result")]
    public JToken Result { get; set; }

    [JsonProperty("elapse")]
    public double Elapse { get; set; }
}

File: framework/kotlin/src/main/kotlin/soda/kotlin/unittest/WorkOutput.kt
package soda.kotlin.unittest

import kotlinx.serialization.Required
import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonNull

@Serializable
class WorkOutput {
    var id: Int = 0
    @Required
    var success: Boolean = false
    var result: JsonElement = JsonNull
    var elapse: Double = 0.0

    fun toJsonString(): String = Json.encodeToString(this)
}
